crypto/crypto/src/noise.rs

```diff
@@ -680,8 +685,7 @@ impl NoiseSession {
         nonce.extend_from_slice(&self.read_nonce.to_be_bytes());
         let nonce = GenericArray::from_slice(&nonce);
 
-        let (buffer, authentication_tag) =
-            ciphertext.split_at_mut(ciphertext.len() - AES_GCM_TAGLEN);
+        let (buffer, authentication_tag) = message.split_at_mut(message.len() - AES_GCM_TAGLEN);
```

Comment:
since these are "encrypt in place" functions having it named ciphertext or plaintext is weird. I can split this commit in several commits if that helps reviews.

Reply:
nah, just a heads up for the future, that way I focus on the important stuff and not get distract... I don't have a good attention span :)